src/core/s3d/include/s3d/disparity/disparities_dense.h

```diff
@@ -10,30 +10,32 @@ namespace s3d {
 
 class DisparitiesDense : public Disparities {
  public:
-  DisparitiesDense(Image disparityMap, DisparityPoint minDisparity, DisparityPoint maxDisparity)
+  DisparitiesDense(Image<uint8_t> disparityMap,
+                   DisparityPoint minDisparity,
+                   DisparityPoint maxDisparity)
       : disparityMap_(std::move(disparityMap)), min_(minDisparity), max_(maxDisparity) {}
 
   const std::vector<DisparityPoint>& getDisparities() override {
     if (!disparities_.empty()) {
       return disparities_;
     }
-    for (int i = 0; i < disparityMap_.mat.rows; ++i) {
-      for (int j = 0; j < disparityMap_.mat.cols; ++j) {
-        auto dx = disparityMap_.mat.at<uint8_t>(i, j);
+    for (int i = 0; i < disparityMap_.height(); ++i) {
+      for (int j = 0; j < disparityMap_.width(); ++j) {
+        auto dx = disparityMap_(i, j);
         disparities_.emplace_back(DisparityPoint({i, j}, {0, dx}));
       }
     }
     return disparities_;
   }
 
-  Image getDisparityMap() override { return disparityMap_; }
+  Image<uint8_t> getDisparityMap() override { return disparityMap_; }
```

Comment:
Member returned by copy. It could be returned by const &. However the base interface would change and maybe that's not always what we want.